x-pack/plugins/lens/public/indexpattern_datasource/operations/definitions/date_histogram.tsx

```diff
@@ -334,3 +347,138 @@ function restrictedInterval(aggregationRestrictions?: Partial<IndexPatternAggRes
+      </EuiPopoverTitle>
+      <EuiText size="s">
+        <FormattedMessage
+          id="xpack.lens.indexPattern.dateHistogram.autoFullDescription"
+          defaultMessage="{shortDescription} {longerDescription} {advancedDescription}"
+          values={{
+            shortDescription: (
+              <p>
+                <FormattedMessage
+                  id="xpack.lens.indexPattern.dateHistogram.autoBasicExplanation"
+                  defaultMessage="Splits a date field into buckets by interval."
+                />
+              </p>
+            ),
+            longerDescription: (
+              <p>
+                <FormattedMessage
+                  id="xpack.lens.indexPattern.dateHistogram.autoLongerExplanation"
+                  defaultMessage="The Lens editor chooses an automatic interval for you by dividing the selected time range by the 
+                  {targetBarSetting} Kibana advanced setting. The calculation tries to present “nice” time interval buckets. The maximum 
+                  number of bars is set by the {maxBarSetting} value."
+                  values={{
+                    maxBarSetting: <EuiCode>{UI_SETTINGS.HISTOGRAM_MAX_BARS}</EuiCode>,
+                    targetBarSetting: <EuiCode>{UI_SETTINGS.HISTOGRAM_BAR_TARGET}</EuiCode>,
+                  }}
+                />
+              </p>
+            ),
+            advancedDescription: (
+              <p>
+                {i18n.translate('xpack.lens.indexPattern.dateHistogram.autoAdvancedExplanation', {
+                  defaultMessage: 'The specific interval adopted follow the logic in this table:',
+                })}
+              </p>
+            ),
+          }}
+        />
+      </EuiText>
+      <EuiBasicTable
+        items={wrapMomentPrecision(() =>
+          search.aggs.boundsDescendingRaw.map(({ bound, interval }) => ({
+            bound:
+              typeof bound === 'number'
+                ? infiniteBound
+                : `${upToLabel} ${humanDurationFormatter.convert(bound)}`,
+            interval: humanDurationFormatter.convert(interval),
+          }))
+        )}
+        columns={[
+          {
+            field: 'bound',
+            name: i18n.translate('xpack.lens.indexPattern.dateHistogram.autoBoundHeader', {
+              defaultMessage: 'Target Interval measured',
+            }),
+          },
+          {
+            field: 'interval',
+            name: i18n.translate('xpack.lens.indexPattern.dateHistogram.autoIntervalHeader', {
+              defaultMessage: 'Interval used',
+            }),
+          },
+        ]}
+      />
+    </EuiPopover>
+  );
+};
+
+// Below 5 seconds the "humanize" call returns the "few seconds" sentence, which is not ok for ms
+// This special config rewrite makes it sure to have precision also for sub-seconds durations
+// ref: https://github.com/moment/moment/issues/348
+function wrapMomentPrecision<T>(callback: () => T): T {
```

Comment:
This logic sets the display to "1 seconds" which should be "1 second" (not plural). Is that a moment issue?

Reply:
Ah, nice catch. I'll investigate

Reply:
Did an investigation on the topic.

**TLDR;** I would leave this glitch as for now, delaying the decision for when `moment.js` will be replaced in the codebase or if the localization of the `humanize` function would be eventually internally handled.

**Longer explanation:**

The issue is relative to `moment.js` as it is reproducible within the `wrapMomentPrecision` function with the following line:

```js
console.log(moment.localeData().relativeTime(1, true, 'ss', false));
```

`moment.js` has an internal `RelativeTimeSpec` configuration that leads to this result, and while it is possible to fix it with a configuration customization, it is rather hard to safely restore the default.
The fix is an override of the configuration with a function that replaces the template string in use to specifically handle the `1 second` case. the problem is that it should be done for all supported locales (~134 in `moment` itself + custom one, not sure how many in Kibana - I've found one custom so far).

The default configuration restoration is hard as there's no (clean) direct way to access the raw `RelativeTimeSpec` object in use, but only access it's output. The thing is not impossible, but `moment.js` typings does not like it: there's an internal prop to be accessed which is not typed and not meant to be accessed.

A potential solution would rely on module augmentation:

```ts
// Need to augment the moment module to access a hidden config parameter
declare module 'moment' {
  interface Locale {
    _config: LocaleSpecification;
  }
}


function wrapMomentPrecision<T>(callback: () => T): T {
  ...

  const defaultLocaleConfig = moment.localeData()._config;
  moment.updateLocale(moment.locale(), {
    relativeTime: {
      // what about other locales than "en" here?
      ss: (n: number): string => n === 1 ? 'a second' : `${n} second`
    },
  });

  // Execute the format/humanize call in the callback
  const result = callback();

  // restore all the default values now in moment to not break it
  moment.updateLocale(moment.locale(), defaultLocaleConfig);
}
```

It would be possible to propose upstream to give access to the `LocaleSpecification` data/configuration (the parent object of `RelativeTimeSpec`) but it is not guaranteed to be merged or accepted (`moment.js` is currently in maintenance mode).
Also it may requires to list and cover all potential locales in Kibana as well.